docs/usage.md

```diff
@@ -44,87 +63,196 @@ TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,
+
+### Other inputs (optional)
+
+Magmap can handle several types of input that can be used for different purpose.
+
+#### Indexes input
+
+The indexes input is used by Sourmash to select genomes that can downloaded in a second step and added to the pipeline.
+It is provided with the `--indexes` parameter and it is a path (local or remote).
+
+```
+https://github.com/nf-core/test-datasets/raw/magmap/testdata/sourmash_test.index.sbt.zip
+```
+
+N.B.: the sbt files can be generated with sourmash (check it [here](https://sourmash.readthedocs.io/en/latest/index.html)) or some can be found in sourmash documentation website.
+E.g. "sourmash gtdb sbt" you will find the indexes for gtdb.
+
+#### metadata input
+
+Magmap accepts several metadata as `.csv` files that provides information about the genomes that you will use in the pipeline.
+
+##### gtdb_metadata
+
+This file contains several information that can be found in gtdb metadata on the official [website](https://data.ace.uq.edu.au/public/gtdb/data/releases/release80/80.0/). You can either use directly the same file found there or you can make a custom one.
+
+##### gtdb-tk metadata
+
+This file contains several information that can be obtained as gtdb-tk output. You can either use directly the same file found there or you can make a custom one.
+
+##### checkM metadata
+
+This file contains several information that can be obtained as checkM output. You can either use directly the same file found there or you can make a custom one.
+
+### Check duplicates
+
+The pipeline is checking if there are any duplicate names among the genomes that the user provides. If there are duplicates, the pipeline will stop and return a file with the contig names that needs to be changed in their name in order to work. This is done to avoid overlapping in the following steps (e.g. same prokka output for the protein sequences and the gffs).
+
+### Filter/remove sequences from the samples (e.g. rRNA sequences with SILVA database)
+
+The pipeline can remove potential contaminants using the BBduk program.
+Specify a fasta file, gzipped or not, with the --sequence_filter sequences.fasta parameter.
+For further documentation, see the [BBduk official website](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/).
+
+```bash
+nextflow run nf-core/magmap -profile docker --outdir results/ --input samples.csv --genomeinfo localgenomes.csv --sequence_filter path/to/file
+```
+
+### Sourmash (optional)
+
+with [Sourmash](https://sourmash.readthedocs.io/en/latest/index.html) you can filter the genomes to be used by magmap in the mapping step. This function is optional but can speed up the process and let you get a better genomes /reads mapping ratio since you are removing all the genomes that are not passing the threshold (that you can select).
```

Comment:
```suggestion
With [Sourmash](https://sourmash.readthedocs.io/en/latest/index.html) you can filter the genomes to be used by magmap in the mapping step. This function is optional but can speed up the process and let you get a better genomes /reads mapping ratio since you are removing all the genomes that are not passing the threshold (that you can select).
```

What threshold? Give an example parameter in the command